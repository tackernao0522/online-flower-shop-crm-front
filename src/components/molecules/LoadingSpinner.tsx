import { Flex, Spinner } from '@chakra-ui/react';
import React from 'react';

interface LoadingSpinnerProps {
  hasMore?: boolean;
  status: string;
}

export const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({
  hasMore,
  status,
}) => {
  if (!hasMore || status !== 'loading') return null;

  return (
    <Flex justify="center" my={4}>
      <Spinner />
    </Flex>
  );
};
