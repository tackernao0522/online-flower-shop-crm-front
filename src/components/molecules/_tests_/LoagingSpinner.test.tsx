import React from 'react';
import { render, screen } from '@testing-library/react';
import { LoadingSpinner } from '../LoadingSpinner';
import '@testing-library/jest-dom';

describe('LoadingSpinner コンポーネント', () => {
  test('hasMoreがfalseの場合、何も表示しない', () => {
    render(<LoadingSpinner hasMore={false} status="loading" />);
    const spinner = screen.queryByText('Loading...');
    expect(spinner).not.toBeInTheDocument();
  });

  test('statusがloading以外の場合、何も表示しない', () => {
    render(<LoadingSpinner hasMore={true} status="idle" />);
    const spinner = screen.queryByText('Loading...');
    expect(spinner).not.toBeInTheDocument();
  });

  test('hasMoreがtrueかつstatusがloadingの場合、Spinnerを表示する', () => {
    render(<LoadingSpinner hasMore={true} status="loading" />);
    const spinner = screen.getByText('Loading...');
    expect(spinner).toBeInTheDocument();
  });

  test('Spinnerが中央に表示される', () => {
    const { container } = render(
      <LoadingSpinner hasMore={true} status="loading" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
