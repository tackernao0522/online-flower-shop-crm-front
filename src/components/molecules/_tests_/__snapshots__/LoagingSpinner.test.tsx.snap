// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner コンポーネント Spinnerが中央に表示される 1`] = `
<div
  class="css-1evii0e"
>
  <div
    class="chakra-spinner css-1y7joxr"
  >
    <span
      class="css-8b45rq"
    >
      Loading...
    </span>
  </div>
</div>
`;
